docs/acceptance-criteria/uacs-tbtc-v2.adoc

```diff
@@ -0,0 +1,291 @@
+| Wallet lifecycle
+| Anyone can send a notification to the wallet registry contract informing that
+  it's time to create a new wallet.
+
+// TODO: Add UAC describing ECDSA client perspective - clients should be
+// programmed to send the notifications when they notice that new wallet
+// creation criteria are met. Do we know how to discourage sending the
+// notifications by multiple clients at the same time?
+
+| Wallet lifecycle - wallet creation
+| Upon processing of the time-to-create-new-wallet notification, the wallet
+  registry contract should determine the validity of the notification.
+
+| Wallet lifecycle - wallet creation
+| When determining the validity of the time-to-create-new-wallet notification,
+  the following aspects should be taken into account: +
+  - is the creation of a new wallet already in progress? (if yes, then
+  notification invalid) +
+  - have at least `wallet_creation_period` blocks passed since the last wallet
+  was created? (if not, then notification invalid) +
+  - are there no wallets or do all wallets have their balances higher or equal
+  `wallet_min_btc`? (if not, then notification invalid).
+
+| Wallet lifecycle - wallet creation
+| If the time-to-create-new-wallet notification is deemed valid, then: +
+  - the notifier should receive reimbursement for the gas spent on the
+  transaction, paid from the DAO-funded ETH pool to the notifier’s address (in
+  the same tx); there should be a limit on the refunded gas price
+  (`max_gas_refund_price`) (if the amount of gwei paid for gas in the DKG result
+  submission is bigger than this value, the submitter will be reimbursed
+  `max_gas_refund_price x gas_used`) +
+  - the wallet registry contract should lock the sortition pool and request a
+  new relay entry from the Random Beacon +
+  - once the value is returned, the wallet registry should start a process of
+  distributed key generation (DKG).
+
+| Wallet lifecycle - wallet creation - DKG
+| At the beginning of the DKG process the DKG contract should emit an event
+  informing about the start of the DKG and the value of current relay entry.
+
+| Wallet lifecycle - wallet creation - DKG - selecting members
+| Upon noticing the DKG start event, all off-chain ECDSA clients should call the
+  sortition function which should select `group_size` (100) pool members to the
+  candidate group based on the current relay entry, ensuring that the higher is
+  the stake of an operator, the higher is his chance of being selected to the
+  group (operators may be chosen to the group multiple times).
+
+// It's not yet clear if any member will be able to submit the result or
+// if there will be some order enforced. Following UAC assumes "any".
+| Wallet lifecycle - wallet creation - DKG
+| After group members are determined, clients should perform off-chain DKG
+  (distributed key generation), resulting in either a success or a timeout. To
+  constitute a success at least `dkg` (90) members of the group must actively
```

Comment:
TODO: In DKGValidator.sol the `dkg` parameter is called `activeThreshold`. There are also some small changes between the naming of other params in the RFC and in the implemented code. It would be good to unify the notation in code and in the docs...